src/ru/nsu/ccfit/zuev/osu/MainScene.java

```diff
@@ -84,7 +84,13 @@
     private ChangeableText musicInfoText;
     //    private ArrayList<BeatmapInfo> beatmaps;
     private Random random = new Random();
-    private Rectangle[] spectrum = new Rectangle[120];
+
+    private Rectangle[] spectrum1 = new Rectangle[120];
+    private Rectangle[] spectrum2 = new Rectangle[120];
+    private Rectangle[] spectrum3 = new Rectangle[120];
+    private Rectangle[] spectrum4 = new Rectangle[120];
+    private Rectangle[] spectrum5 = new Rectangle[120];
+
```

Comment:
Do not use this way to draw so many objects, in AndEngine those codes means 120*5 draw calls, it's not efficiently.
You can use [com.edlplan.framework.support](https://github.com/osudroid/osu-droid/tree/master/src/com/edlplan/framework/support), which is much faster.

Reply:
I would have to time with your framework first, what I can do for now would be to reduce it?

Reply:
you can close this pull request because if you can't totally solve this problem, i won't approve this pull request.